dbms/src/Functions/FunctionsString.cpp

```diff
@@ -5022,52 +5025,443 @@ class FunctionPosition : public IFunction
+        const ColumnString::Offsets & col0_offsets,
+        const String & col1_str,
+        PaddedPODArray<Int64> & res) const
+    {
+        size_t row_num = col0_offsets.size();
+        ColumnString::Offset prev_col0_str_offset = 0;
+
+        String col0_container;
+        String col1_container;
+        std::vector<size_t> lens;
+        StringRef col1_collation_str;
+
+        if (collator == nullptr || collator->isPaddingBinary())
+            col1_collation_str = fastPath<true>(col1_str.data(), col1_str.size(), &lens);
+        else
+            col1_collation_str = collator->convert(col1_str.data(), col1_str.size(), col1_container, &lens);
+
+        for (size_t i = 0; i < row_num; i++)
+        {
+            size_t col0_str_len = col0_offsets[i] - prev_col0_str_offset - 1;
+
+            if unlikely (col0_str_len == 0)
+            {
+                res[i] = 1;
+            }
+            else
+            {
+                StringRef col0_collation_str;
+
+                if (collator == nullptr || collator->isPaddingBinary())
+                    col0_collation_str = fastPath<false>(
+                        reinterpret_cast<const char *>(&col0_data[prev_col0_str_offset]),
+                        col0_str_len,
+                        nullptr);
+                else
+                    col0_collation_str = collator->sortKeyNoTrim(
+                        reinterpret_cast<const char *>(&col0_data[prev_col0_str_offset]),
+                        col0_str_len,
+                        col0_container);
+
+                searchAndFillResult(
+                    i,
+                    col0_collation_str.data,
+                    col0_collation_str.size,
+                    col1_collation_str.data,
+                    col1_collation_str.size,
+                    lens,
+                    res);
+            }
+            prev_col0_str_offset = col0_offsets[i];
+        }
+    }
+
+    void constVector(
+        const String & col0_str,
+        const ColumnString::Chars_t & col1_data,
+        const ColumnString::Offsets & col1_offsets,
+        PaddedPODArray<Int64> & res) const
+    {
+        size_t row_num = col1_offsets.size();
+        ColumnString::Offset prev_col1_str_offset = 0;
+
+        String col0_container;
+        String col1_container;
+        std::vector<size_t> lens;
+        StringRef col0_collation_str;
+
+        if (collator == nullptr || collator->isPaddingBinary())
+            col0_collation_str = fastPath<false>(col0_str.c_str(), col0_str.size(), nullptr);
+        else
+            col0_collation_str = collator->sortKeyNoTrim(col0_str.c_str(), col0_str.size(), col0_container);
+
+        for (size_t i = 0; i < row_num; i++)
+        {
+            size_t col1_str_len = col1_offsets[i] - prev_col1_str_offset - 1;
+
+            StringRef col1_collation_str;
+            if (collator == nullptr || collator->isPaddingBinary())
+                col1_collation_str = fastPath<true>(
```

Comment:
I think just call `TiDB::convertForBinCollator` is ok?

Reply:
> I think just call `TiDB::convertForBinCollator` is ok?

I think it's better to unify them with a function.